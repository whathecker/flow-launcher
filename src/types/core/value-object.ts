export type PriorityTier = "n/a" | "highest" | "high" | "mid" | "low";

export type PriorityMeasure = "n/a" | "yes" | "no"; // TODO: consider to use MIXED type (boolean and null) instead?

export class Priority {
  constructor(
    public tier: PriorityTier,
    public importance: PriorityMeasure,
    public urgency: PriorityMeasure,
  ) {}
}
