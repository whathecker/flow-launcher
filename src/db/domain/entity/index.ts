import { Priority } from "../value-object";

export type ID = number | string;

export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public id: ID,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
  ) {}
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
    public id?: ID,
  ) {}
}
