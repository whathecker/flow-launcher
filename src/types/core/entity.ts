import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export interface IAddTaskInput {
  title: string;
  description: string;
  goal_id: string;
}

export interface IFetchTasksInput {
  goal_id: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public status: TaskStatus,
    public description?: string,
    public priority?: Priority,
  ) {}
}
