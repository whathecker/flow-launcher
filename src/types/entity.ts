import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[] | [],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
  ) {}
}
