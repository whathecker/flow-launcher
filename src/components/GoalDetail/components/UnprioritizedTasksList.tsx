import React from "react";
import { StyleSheet } from "react-native";
import { View, Text, Touchable } from "../../Themed";
import UnprioritizedTask from "./UnprioritizedTask";
import { Container, Typography, Color } from "../../../styles";
import { Task } from "../../../types/core/entity";
import { navigationRef } from "../../../utils";

type UnprioritizedTasksProp = {
  tasks: Task[];
};

const UnprioritizedTasksList: React.FC<UnprioritizedTasksProp> = ({
  tasks,
}: UnprioritizedTasksProp) => {
  return (
    <View style={styles.wrapper}>
      {tasks[0] ? <UnprioritizedTask title={tasks[0].title} /> : null}
      {tasks[1] ? <UnprioritizedTask title={tasks[1].title} /> : null}
      {tasks[2] ? <UnprioritizedTask title={tasks[2].title} /> : null}
      {tasks.length > 3 ? (
        <View style={styles.moreTasksIndicatorWrapper}>
          <Text>{`...`}</Text>
        </View>
      ) : null}
      {tasks.length > 0 ? (
        <View style={styles.buttonAreaWrapper}>
          <Touchable>
            <Text style={styles.sellAllBtn}>{`See All (90)`}</Text>
          </Touchable>
          <Touchable
            onPress={() => {
              navigationRef.navigate("Prior");
            }}
          >
            <Text style={styles.prioritizeBtn}>{`Start Prioritizing`}</Text>
          </Touchable>
        </View>
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    ...Container.centerAlignedVertical,
    width: "90%",
    marginLeft: "5%",
    marginRight: "5%",
    marginTop: 15,
    marginBottom: 8,
  },
  moreTasksIndicatorWrapper: {
    marginTop: 5,
    marginBottom: 5,
  },
  buttonAreaWrapper: {
    ...Container.flexStart,
    justifyContent: "space-between",
    width: "100%",
    paddingTop: 5,
  },
  prioritizeBtn: {
    ...Typography.p,
    textDecorationLine: "underline",
    color: Color.light.primaryBtn,
    fontSize: 18,
  },
  sellAllBtn: {
    ...Typography.p,
    textDecorationLine: "underline",
    color: Color.light.text,
    fontSize: 18,
  },
});

export default UnprioritizedTasksList;
