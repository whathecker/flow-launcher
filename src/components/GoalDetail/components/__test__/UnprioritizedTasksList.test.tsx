import * as React from "react";
import renderer from "react-test-renderer";
import UnprioritizedTasksList from "../UnprioritizedTasksList";
import { Task } from "../../../../types/core/entity";

describe("Test UnprioritizedTasksList component", () => {
  const dummayTasks: Task[] = [
    {
      _id: "sjdiodfi103jsldicsd",
      goal_id: "asjdklasf0qwdakddasd",
      title: "test tasks",
      status: "open",
    },
    {
      _id: "sjdiodfi103jsldicsdsd",
      goal_id: "asjdklasf0qwdakddasd",
      title: "test tasks -1",
      status: "open",
    },
    {
      _id: "sjdiodfi103jsldicsdasdadgd",
      goal_id: "asjdklasf0qwdakddasd",
      title: "test tasks -2",
      status: "open",
    },
    {
      _id: "sjdiodfi103jsldicsdadasdasfwefewvfwe",
      goal_id: "asjdklasf0qwdakddasd",
      title: "test tasks-3",
      status: "open",
    },
  ];

  it("Should renders UnprioritizedTasksList correctly", () => {
    const tree = renderer
      .create(<UnprioritizedTasksList tasks={dummayTasks} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
