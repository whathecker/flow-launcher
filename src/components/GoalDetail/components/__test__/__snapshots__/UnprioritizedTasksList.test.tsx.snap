// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UnprioritizedTasksList component Should renders UnprioritizedTasksList correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginBottom": 8,
        "marginLeft": "5%",
        "marginRight": "5%",
        "marginTop": 15,
        "width": "90%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "borderColor": "#848484",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 5,
          "paddingBottom": 5,
          "paddingTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 14,
          },
        ]
      }
    >
      test tasks
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "borderColor": "#848484",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 5,
          "paddingBottom": 5,
          "paddingTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 14,
          },
        ]
      }
    >
      test tasks -1
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "borderColor": "#848484",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 5,
          "paddingBottom": 5,
          "paddingTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 14,
          },
        ]
      }
    >
      test tasks -2
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "marginBottom": 5,
          "marginTop": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          undefined,
        ]
      }
    >
      ...
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 30,
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "fontFamily": "Lexend-Regular",
              "fontSize": 14,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        See All (90)
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "color": "#0FA858",
              "fontFamily": "Lexend-Regular",
              "fontSize": 14,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Start Prioritizing
      </Text>
    </View>
  </View>
</View>
`;
